sdk/dynamicconfig/metric/controller/push/period_matcher.go

```diff
@@ -0,0 +1,130 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package push
+
+import (
+	"sync"
+	"time"
+
+	pb "github.com/open-telemetry/opentelemetry-proto/gen/go/experimental/metricconfigservice"
+	"go.opentelemetry.io/contrib/sdk/dynamicconfig/internal/metricpattern"
+	"go.opentelemetry.io/contrib/sdk/dynamicconfig/metric"
+)
+
+const tolerance float64 = 0.1
+
+type PeriodMatcher struct {
+	metrics   map[string]*CollectData
+	startTime time.Time
+
+	m     sync.Mutex
+	sched []*pb.MetricConfigResponse_Schedule
+}
+
+type CollectData struct {
+	lastCollected time.Time
+	period        time.Duration
+}
+
+func (matcher *PeriodMatcher) MarkStart(startTime time.Time) {
+	matcher.startTime = startTime
+}
+
+func (matcher *PeriodMatcher) ConsumeSchedules(sched []*pb.MetricConfigResponse_Schedule) {
```

Comment:
ResetSchedules()?

Reply:
Interesting thought, I think the original name was something like `Reset`, but we're not quite resetting the schedules so much as processing a new set of schedules -- hence I chose the verb `consume` rather than `reset`

Reply:
Why do you think this is not resetting the schedules but processing a new set of schedules?

matcher.sched = sched

Suggest it is setting the sched to a different set of schedules, which is the definition of reset as "set again or differently", right?

Reply:
To elaborate a little further, the method also initializes the `sched` and `metrics` fields, during the initial gRPC call to obtain schedule data. In this way, I chose `consume` to more generally refer to processing new data, rather than `reset` which has the connotation that the data already exists. Though if `reset` is more broadly intuitive, I can go ahead and change it if you like?

Reply:
As discussed, let's change to ApplySchedules or SetSchedules. Let me know once you are done with that.